pandas/tests/io/json/test_json_table_schema_ext_dtype.py

```diff
@@ -0,0 +1,337 @@
+                self._month[i] = int(m)
+                self._day[i] = int(d)
+
+        else:
+            raise TypeError(f"{type(dates)} is not supported")
+
+    @property
+    def dtype(self) -> ExtensionDtype:
+        return DateDtype()
+
+    def astype(self, dtype, copy=True):
+        dtype = pandas_dtype(dtype)
+
+        if isinstance(dtype, DateDtype):
+            data = self.copy() if copy else self
+        else:
+            data = self.to_numpy(dtype=dtype, copy=copy, na_value=dt.date.min)
+
+        return data
+
+    @property
+    def nbytes(self) -> int:
+        return self._year.nbytes + self._month.nbytes + self._day.nbytes
+
+    def __len__(self) -> int:
+        return len(self._year)  # all 3 arrays are enforced to have the same length
+
+    def __getitem__(self, item: PositionalIndexer):
+        if isinstance(item, int):
+            return dt.date(self._year[item], self._month[item], self._day[item])
+        else:
+            raise NotImplementedError("only ints are supported as indexes")
+
+    def __setitem__(self, key: Union[int, slice, np.ndarray], value: Any):
+        if not isinstance(key, int):
+            raise NotImplementedError("only ints are supported as indexes")
+
+        if not isinstance(value, dt.date):
+            raise TypeError("you can only set datetime.date types")
+
+        self._year[key] = value.year
+        self._month[key] = value.month
+        self._day[key] = value.day
+
+    def __repr__(self) -> str:
+        return f"DateArray{list(zip(self._year, self._month, self._day))}"
+
+    def copy(self) -> "DateArray":
+        return DateArray((self._year.copy(), self._month.copy(), self._day.copy()))
+
+    def isna(self) -> np.ndarray:
+        return np.logical_and(
+            np.logical_and(
+                self._year == dt.date.min.year, self._month == dt.date.min.month
+            ),
+            self._day == dt.date.min.day,
+        )
+
+    @classmethod
+    def _from_sequence(cls, scalars, *, dtype: Optional[Dtype] = None, copy=False):
+        if isinstance(scalars, dt.date):
+            pass
+        elif isinstance(scalars, DateArray):
+            pass
+        elif isinstance(scalars, np.ndarray):
+            scalars = scalars.astype("U10")  # 10 chars for yyyy-mm-dd
+            return DateArray(scalars)
+
+
+class TestBuildSchema:
+    def setup_method(self, method):
+        self.da = DateArray([dt.date(2021, 10, 10)])
+        self.dc = DecimalArray([decimal.Decimal(10)])
+        self.df = DataFrame(
+            {
+                "A": self.da,
+                "B": self.dc,
+            }
+        )
+
+    def test_build_table_schema(self):
+        result = build_table_schema(self.df, version=False)
+        expected = {
+            "fields": [
+                {"name": "index", "type": "integer"},
+                {"name": "A", "type": "any", "extDtype": "DateDtype"},
+                {"name": "B", "type": "any", "extDtype": "decimal"},
```

Comment:
Can you add some of the built-in extension dtypes, e.g. `String` and `Int64` ? (unless those were already handled in the other test file)

Reply:
I'll check but I think they're handled in the other test.

Reply:
The array type for `Int64` & friends is tested:
https://github.com/pandas-dev/pandas/blob/0b0cac5c46506257b0195d78576c0de4fbb55636/pandas/tests/arrays/test_array.py#L38
`String` is here:
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/arrays/string_/test_string.py

Reply:
No, I mean that you want to test `String` and `Int64` in the context of `to_json` and `read_json` .  What you pointed to are the regular tests for those dtypes.

Reply:
Added in the last commit